diff --git a/litesdcard/phy.py b/litesdcard/phy.py
index 53a7143..13edb55 100644
--- a/litesdcard/phy.py
+++ b/litesdcard/phy.py
@@ -521,11 +521,20 @@ class SDPHYIOGen(SDPHYIO):
             self.comb += pads.rst.eq(0)
 
         # Clk
-        self.specials += SDROutput(
-            clk = ClockSignal("sys"),
-            i   = clocker.clk,
-            o   = pads.clk
-        )
+        clk = Signal()
+        self.specials += [
+            SDROutput(
+                clk = ClockSignal("sys"),
+                i   = clocker.clk,
+                o   = clk,
+            ),
+            Instance("DELAYG",
+                p_DEL_MODE  = "USER_DEFINED",
+                p_DEL_VALUE = 120, # (25ps per tap)
+                i_A         = clk,
+                o_Z         = pads.clk
+            )
+        ]
 
         # Cmd
         self.specials += SDRTristate(
